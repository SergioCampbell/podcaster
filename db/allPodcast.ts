import { AllPodcasts } from "@/interface/podcasts";

export const allPodcast: AllPodcasts = [
  {
    "im:name": {
      label: "Mortal Sin",
    },
    "im:image": [
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts126/v4/73/e4/8e/73e48e46-2812-f78f-e571-454e85b59be7/mza_4549916304669920567.jpg/55x55bb.png",
        attributes: {
          height: "55",
        },
      },
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts126/v4/73/e4/8e/73e48e46-2812-f78f-e571-454e85b59be7/mza_4549916304669920567.jpg/60x60bb.png",
        attributes: {
          height: "60",
        },
      },
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts126/v4/73/e4/8e/73e48e46-2812-f78f-e571-454e85b59be7/mza_4549916304669920567.jpg/170x170bb.png",
        attributes: {
          height: "170",
        },
      },
    ],
    summary: {
      label:
        "Dawn and Nick Hacheney are the perfect couple: spiritual, loving, and devoted to the church where Nick is a pastor. When Dawn is killed in a house fire the day after Christmas, the pastor and his flock are devastated. What few knew at the time was the dark prophecy that foretold it. The latest podcast from Dateline and Josh Mankiewicz is about sex, lies, religion… and murder.\n\nFollow now to get the latest episodes of ‘Mortal Sin’ completely free, or subscribe to Dateline Premium on Apple Podcasts for early access and ad-free listening: apple.co/datelinepremium",
    },
    "im:price": {
      label: "Get",
      attributes: {
        amount: "0",
        currency: "USD",
      },
    },
    "im:contentType": {
      attributes: {
        term: "Podcast",
        label: "Podcast",
      },
    },
    rights: {
      label: "© 2023 NBCUniversal Media LLC, all rights reserved",
    },
    title: {
      label: "Mortal Sin - NBC News",
    },
    link: {
      attributes: {
        rel: "alternate",
        type: "text/html",
        href: "https://podcasts.apple.com/us/podcast/mortal-sin/id1718562878?uo=2",
      },
    },
    id: {
      label:
        "https://podcasts.apple.com/us/podcast/mortal-sin/id1718562878?uo=2",
      attributes: {
        "im:id": "1718562878",
      },
    },
    "im:artist": {
      label: "NBC News",
    },
    category: {
      attributes: {
        "im:id": "1488",
        term: "True Crime",
        scheme:
          "https://podcasts.apple.com/us/genre/podcasts-true-crime/id1488?uo=2",
        label: "True Crime",
      },
    },
    "im:releaseDate": {
      label: "2023-12-19T03:00:00-07:00",
      attributes: {
        label: "December 19, 2023",
      },
    },
  },
  {
    "im:name": {
      label: "Who shat on the floor at my wedding?",
    },
    "im:image": [
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts116/v4/d1/f4/ae/d1f4ae64-32af-d8a5-9531-07bd8bef2a05/mza_3539549108746744251.jpeg/55x55bb.png",
        attributes: {
          height: "55",
        },
      },
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts116/v4/d1/f4/ae/d1f4ae64-32af-d8a5-9531-07bd8bef2a05/mza_3539549108746744251.jpeg/60x60bb.png",
        attributes: {
          height: "60",
        },
      },
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts116/v4/d1/f4/ae/d1f4ae64-32af-d8a5-9531-07bd8bef2a05/mza_3539549108746744251.jpeg/170x170bb.png",
        attributes: {
          height: "170",
        },
      },
    ],
    summary: {
      label:
        "A true story involving two brides and one turd. What do you do if someone defecates on the floor at your wedding? Re-evaluate your friends and family? No, you launch an investigation to find who did it. Join the brides, Helen Mclaughlin and Karen Whitehouse, and the extremely under-qualified “Detective” Lauren Kilby, as they interrogate wedding guests, hook bridesmaids up to polygraph machines and speak with top forensic experts in an attempt to crack the case. Because crime is not a laughing matter. Well, maybe except for this one.\n\n\n\n Hosted on Acast. See acast.com/privacy for more information.",
    },
    "im:price": {
      label: "Get",
      attributes: {
        amount: "0",
        currency: "USD",
      },
    },
    "im:contentType": {
      attributes: {
        term: "Podcast",
        label: "Podcast",
      },
    },
    rights: {
      label: "© Who shat on the floor at my wedding?",
    },
    title: {
      label:
        "Who shat on the floor at my wedding? - Who shat on the floor at my wedding?",
    },
    link: {
      attributes: {
        rel: "alternate",
        type: "text/html",
        href: "https://podcasts.apple.com/us/podcast/who-shat-on-the-floor-at-my-wedding/id1538165808?uo=2",
      },
    },
    id: {
      label:
        "https://podcasts.apple.com/us/podcast/who-shat-on-the-floor-at-my-wedding/id1538165808?uo=2",
      attributes: {
        "im:id": "1538165808",
      },
    },
    "im:artist": {
      label: "Who shat on the floor at my wedding?",
    },
    category: {
      attributes: {
        "im:id": "1303",
        term: "Comedy",
        scheme:
          "https://podcasts.apple.com/us/genre/podcasts-comedy/id1303?uo=2",
        label: "Comedy",
      },
    },
    "im:releaseDate": {
      label: "2021-03-31T01:47:00-07:00",
      attributes: {
        label: "March 31, 2021",
      },
    },
  },
  {
    "im:name": {
      label: "On Musk with Walter Isaacson",
    },
    "im:image": [
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts116/v4/e2/30/9c/e2309c86-d54f-1c85-2121-67c7104441df/mza_3219819313662797266.jpg/55x55bb.png",
        attributes: {
          height: "55",
        },
      },
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts116/v4/e2/30/9c/e2309c86-d54f-1c85-2121-67c7104441df/mza_3219819313662797266.jpg/60x60bb.png",
        attributes: {
          height: "60",
        },
      },
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts116/v4/e2/30/9c/e2309c86-d54f-1c85-2121-67c7104441df/mza_3219819313662797266.jpg/170x170bb.png",
        attributes: {
          height: "170",
        },
      },
    ],
    summary: {
      label:
        "What is it like to shadow Elon Musk for two years? To sit courtside as he builds a rocket? Or tears apart an engineer? Or couch surfs at the homes of billionaires? And how on earth do you make sense of it all? Walter Isaacson is the biographer of giants: DaVinci, Franklin, Doudna, Jobs...and now Musk, former enfante terrible, rocket launcher, electric car innovator, and Twitter—er, X—disruptor, to put it gently. In this four part series, author Evan Ratliff (Mastermind, Longform Podcast) sits down with Isaacson to draw out the behind-the-scenes stories of this epic biography, and what the writer has learned as an outsider inside Silicon Valley.  ",
    },
    "im:price": {
      label: "Get",
      attributes: {
        amount: "0",
        currency: "USD",
      },
    },
    "im:contentType": {
      attributes: {
        term: "Podcast",
        label: "Podcast",
      },
    },
    rights: {
      label: "© 2023 iHeartPodcasts",
    },
    title: {
      label: "On Musk with Walter Isaacson - iHeartPodcasts and Kaleidoscope",
    },
    link: {
      attributes: {
        rel: "alternate",
        type: "text/html",
        href: "https://podcasts.apple.com/us/podcast/on-musk-with-walter-isaacson/id1719795462?uo=2",
      },
    },
    id: {
      label:
        "https://podcasts.apple.com/us/podcast/on-musk-with-walter-isaacson/id1719795462?uo=2",
      attributes: {
        "im:id": "1719795462",
      },
    },
    "im:artist": {
      label: "iHeartPodcasts and Kaleidoscope",
    },
    category: {
      attributes: {
        "im:id": "1324",
        term: "Society & Culture",
        scheme:
          "https://podcasts.apple.com/us/genre/podcasts-society-culture/id1324?uo=2",
        label: "Society & Culture",
      },
    },
    "im:releaseDate": {
      label: "2023-12-12T02:05:00-07:00",
      attributes: {
        label: "December 12, 2023",
      },
    },
  },
  {
    "im:name": {
      label: "The Daily",
    },
    "im:image": [
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts115/v4/1c/ac/04/1cac0421-4483-ff09-4f80-19710d9feda4/mza_12421371692158516891.jpeg/55x55bb.png",
        attributes: {
          height: "55",
        },
      },
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts115/v4/1c/ac/04/1cac0421-4483-ff09-4f80-19710d9feda4/mza_12421371692158516891.jpeg/60x60bb.png",
        attributes: {
          height: "60",
        },
      },
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts115/v4/1c/ac/04/1cac0421-4483-ff09-4f80-19710d9feda4/mza_12421371692158516891.jpeg/170x170bb.png",
        attributes: {
          height: "170",
        },
      },
    ],
    summary: {
      label:
        "This is what the news should sound like. The biggest stories of our time, told by the best journalists in the world. Hosted by Michael Barbaro and Sabrina Tavernise. Twenty minutes a day, five days a week, ready by 6 a.m.\n\nListen to this podcast in New York Times Audio, our new iOS app for news subscribers. Download now at nytimes.com/audioapp",
    },
    "im:price": {
      label: "Get",
      attributes: {
        amount: "0",
        currency: "USD",
      },
    },
    "im:contentType": {
      attributes: {
        term: "Podcast",
        label: "Podcast",
      },
    },
    rights: {
      label:
        "© 2020-2021 THE NEW YORK TIMES COMPANY; The New York Times encourages the use of RSS feeds for personal use in a news reader or as part of a non-commercial blog, subject to your agreement to our Terms of Service.",
    },
    title: {
      label: "The Daily - The New York Times",
    },
    link: {
      attributes: {
        rel: "alternate",
        type: "text/html",
        href: "https://podcasts.apple.com/us/podcast/the-daily/id1200361736?uo=2",
      },
    },
    id: {
      label:
        "https://podcasts.apple.com/us/podcast/the-daily/id1200361736?uo=2",
      attributes: {
        "im:id": "1200361736",
      },
    },
    "im:artist": {
      label: "The New York Times",
      attributes: {
        href: "https://podcasts.apple.com/us/artist/the-new-york-times/121664449?uo=2",
      },
    },
    category: {
      attributes: {
        "im:id": "1526",
        term: "Daily News",
        scheme:
          "https://podcasts.apple.com/us/genre/podcasts-news-daily-news/id1526?uo=2",
        label: "Daily News",
      },
    },
    "im:releaseDate": {
      label: "2023-12-19T02:45:00-07:00",
      attributes: {
        label: "December 19, 2023",
      },
    },
  },
  {
    "im:name": {
      label: "The Ramsey Show",
    },
    "im:image": [
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts116/v4/2a/db/c7/2adbc7d6-dcde-d8ae-481e-7f4826abf37a/mza_12577097097957624675.jpg/55x55bb.png",
        attributes: {
          height: "55",
        },
      },
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts116/v4/2a/db/c7/2adbc7d6-dcde-d8ae-481e-7f4826abf37a/mza_12577097097957624675.jpg/60x60bb.png",
        attributes: {
          height: "60",
        },
      },
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts116/v4/2a/db/c7/2adbc7d6-dcde-d8ae-481e-7f4826abf37a/mza_12577097097957624675.jpg/170x170bb.png",
        attributes: {
          height: "170",
        },
      },
    ],
    summary: {
      label:
        "The Ramsey Show believes you can build wealth and take control of your life—no matter what stupid mistakes you've made with money. Join as Dave Ramsey and his team of experts answer your questions on the top problems holding you back. Listen now or ask your question live by calling 888.825.5225 weekdays from 2–5 p.m. ET. Learn more at www.ramseysolutions.com",
    },
    "im:price": {
      label: "Get",
      attributes: {
        amount: "0",
        currency: "USD",
      },
    },
    "im:contentType": {
      attributes: {
        term: "Podcast",
        label: "Podcast",
      },
    },
    rights: {
      label: "© 2023",
    },
    title: {
      label: "The Ramsey Show - Ramsey Network",
    },
    link: {
      attributes: {
        rel: "alternate",
        type: "text/html",
        href: "https://podcasts.apple.com/us/podcast/the-ramsey-show/id77001367?uo=2",
      },
    },
    id: {
      label:
        "https://podcasts.apple.com/us/podcast/the-ramsey-show/id77001367?uo=2",
      attributes: {
        "im:id": "77001367",
      },
    },
    "im:artist": {
      label: "Ramsey Network",
      attributes: {
        href: "https://podcasts.apple.com/us/artist/ramsey-network/834566130?uo=2",
      },
    },
    category: {
      attributes: {
        "im:id": "1412",
        term: "Investing",
        scheme:
          "https://podcasts.apple.com/us/genre/podcasts-business-investing/id1412?uo=2",
        label: "Investing",
      },
    },
    "im:releaseDate": {
      label: "2023-12-18T15:06:00-07:00",
      attributes: {
        label: "December 18, 2023",
      },
    },
  },
  {
    "im:name": {
      label: "Crime Junkie",
    },
    "im:image": [
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts126/v4/8c/35/04/8c350430-2fbf-98d0-0a25-00b76550ffeb/mza_13445204151221888086.jpg/55x55bb.png",
        attributes: {
          height: "55",
        },
      },
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts126/v4/8c/35/04/8c350430-2fbf-98d0-0a25-00b76550ffeb/mza_13445204151221888086.jpg/60x60bb.png",
        attributes: {
          height: "60",
        },
      },
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts126/v4/8c/35/04/8c350430-2fbf-98d0-0a25-00b76550ffeb/mza_13445204151221888086.jpg/170x170bb.png",
        attributes: {
          height: "170",
        },
      },
    ],
    summary: {
      label:
        "If you can never get enough true crime... Congratulations, you’ve found your people.",
    },
    "im:price": {
      label: "Get",
      attributes: {
        amount: "0",
        currency: "USD",
      },
    },
    "im:contentType": {
      attributes: {
        term: "Podcast",
        label: "Podcast",
      },
    },
    rights: {
      label: "© audiochuck",
    },
    title: {
      label: "Crime Junkie - audiochuck",
    },
    link: {
      attributes: {
        rel: "alternate",
        type: "text/html",
        href: "https://podcasts.apple.com/us/podcast/crime-junkie/id1322200189?uo=2",
      },
    },
    id: {
      label:
        "https://podcasts.apple.com/us/podcast/crime-junkie/id1322200189?uo=2",
      attributes: {
        "im:id": "1322200189",
      },
    },
    "im:artist": {
      label: "audiochuck",
      attributes: {
        href: "https://podcasts.apple.com/us/artist/audiochuck/1485045052?uo=2",
      },
    },
    category: {
      attributes: {
        "im:id": "1488",
        term: "True Crime",
        scheme:
          "https://podcasts.apple.com/us/genre/podcasts-true-crime/id1488?uo=2",
        label: "True Crime",
      },
    },
    "im:releaseDate": {
      label: "2023-12-11T00:00:00-07:00",
      attributes: {
        label: "December 11, 2023",
      },
    },
  },
];
