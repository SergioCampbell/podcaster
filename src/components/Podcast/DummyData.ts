export const dummyData = () => {
    const items = [];
    for (let i = 0; i < 10; i++) {
      items.push({
        id: i,
        title: `Item ${i}`,
        date: 100,
        duration: 1,
      });
    }
    return items;
   }
