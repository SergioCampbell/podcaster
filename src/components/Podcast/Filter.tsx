"use client";

import { Box, Chip, Stack, TextField } from "@mui/material";
import { useState, useEffect, ChangeEvent } from "react";

interface FilterPros {
  searchQueary: string;
  setSearchQueary: (value: string) => void;
  totalPodcasts: number;
}
export const Filter = ({
  searchQueary,
  setSearchQueary,
  totalPodcasts,
}: FilterPros) => {
  const [myWidth, setMyWidth] = useState(0);

  const handleSearch = (e: ChangeEvent<HTMLInputElement>) => {
    const value: string = e.currentTarget.value;
    setSearchQueary(value);
  };

  useEffect(() => {
    setMyWidth(window.screen.width);
  }, []);
  return (
    <Stack
      component="search"
      display={myWidth < 400 ? "contents" : "flex"}
      direction="row-reverse"
      maxWidth={myWidth < 400 ? "100%" : "98.5%"}
      sx={{ mt: 3 }}
    >
      <Box component="form" name="searchPodcast" mt={3}>
        <TextField
          name="search-podcast"
          id="filter-podcast"
          data-testid="search-podcast"
          label=""
          variant="outlined"
          placeholder="Filter Podcasts..."
          size="small"
          fullWidth
          value={searchQueary}
          onChange={handleSearch}
        />
      </Box>
      <Box component={"span"} mt={3.3} mr={1}>
        {totalPodcasts > 0 ? (
          <Chip
            label={totalPodcasts}
            color="primary"
            sx={{ alignContent: "center" }}
          />
        ) : null}
      </Box>
    </Stack>
  );
};
