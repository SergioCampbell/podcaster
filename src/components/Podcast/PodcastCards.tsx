"use client";
import Link from "next/link";
import { AllPodcasts } from "@/interface/podcasts";
import {
  Card,
  CardContent,
  Typography,
  Skeleton,
  Avatar,
  Box,
  Tooltip,
} from "@mui/material";

interface Props {
  allPodcast: AllPodcasts;
  loading?: boolean;
}
/**
 * Renders a list of podcast cards.
 *
 * @param {Props} allPodcast - The array of podcast data.
 * @param {boolean} loading - Indicates whether the podcast data is still loading.
 * @return {JSX.Element} The JSX element representing the rendered podcast cards.
 */
export const PodcastCards = ({ allPodcast, loading }: Props) => {
  return (
    <>
      {allPodcast.map((podcast) => (
        <Box key={podcast.id?.attributes?.["im:id"]}>
          <Link
            style={{ color: "white", textDecoration: "none" }}
            href="/podcast/[podcastid]"
            as={`/podcast/${podcast.id?.attributes?.["im:id"]}`}
          >
            <Card
              sx={{
                marginTop: 3,
                marginBottom: 2,
                boxShadow: 5,
                maxWidth: 300,
                maxHeight: 300,
              }}
            >
              <CardContent>
                {loading ? (
                  <Skeleton
                    animation="wave"
                    variant="circular"
                    width={150}
                    height={150}
                  />
                ) : (
                  <Avatar
                    sx={{
                      width: 150,
                      height: 150,
                      left: 50,
                      bottom: 10,
                    }}
                    alt={podcast.title?.label}
                    src={podcast?.["im:image"]?.[2]?.label || ""}
                  />
                )}
                {loading ? (
                  <>
                    <Skeleton
                      animation="wave"
                      variant="text"
                      component={"h5"}
                    />
                    <Skeleton animation="wave" variant="text" />
                  </>
                ) : (
                  <>
                    <Tooltip title={podcast.title?.label}>
                      <Typography
                        gutterBottom
                        variant="h5"
                        component="h5"
                        sx={{
                          overflow: "hidden",
                          textOverflow: "ellipsis",
                          whiteSpace: "nowrap",
                          maxLines: 2,
                        }}
                      >
                        {podcast.title?.label}
                      </Typography>
                    </Tooltip>
                    <Typography gutterBottom sx={{ fontStyle: "italic" }}>
                      {podcast["im:artist"]?.label}
                    </Typography>
                  </>
                )}
              </CardContent>
            </Card>
          </Link>
        </Box>
      ))}
    </>
  );
};
