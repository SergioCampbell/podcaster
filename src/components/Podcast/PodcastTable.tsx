"use client";

import { Result } from "@/interface/podcastDetail";
import { TABLE_HEADER } from "@/libs/constans";
import {
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
} from "@mui/material";
import {
  getCoreRowModel,
  useReactTable,
  flexRender,
  getSortedRowModel,
} from "@tanstack/react-table";
import type { ColumnDef, SortingState } from "@tanstack/react-table";
import Link from "next/link";
import { useParams } from "next/navigation";
import { useState } from "react";

interface ReactTableProps<T extends object> {
  data: Result[];
}

export const PodcastTable = <T extends object>({
  data,
}: ReactTableProps<T>) => {
  const { podcastId } = useParams();

  const columns: ColumnDef<Result>[] = [
    {
      header: TABLE_HEADER.Title,
      accessorKey: "trackName",
      cell: ({ row }) => {
        const episodeId = row.original.trackId;
        return (
          <Link
            href="/podcast/[podcastId]/episode/[episodeId]"
            as={`/podcast/${podcastId}/episode/${episodeId}`}
            style={{ textDecoration: "none" }}
          >
            {row.original.trackName}
          </Link>
        );
      },
    },
    {
      header: TABLE_HEADER.Date,
      cell: ({ row }) => {
        const date = new Date(row.original.releaseDate);
        return date.toLocaleDateString("es-ES");
      },
      accessorKey: "releaseDate",
    },
    {
      header: TABLE_HEADER.Duration,
      cell: ({ row }) => {
        const minutes = Math.floor(row.original.trackTimeMillis / 60000);
        const seconds = Math.floor(
          (row.original.trackTimeMillis % 60000) / 1000
        );
        return `${minutes}:${seconds.toString().padStart(2, "0")}`;
      },
      accessorKey: "trackTimeMillis",
    },
  ];
  const [sorting, setSorting] = useState<SortingState>([]);
  const table = useReactTable({
    data,
    columns,
    state: {
      sorting,
    },
    onSortingChange: setSorting,
    getCoreRowModel: getCoreRowModel(),
    getSortedRowModel: getSortedRowModel(),
    debugTable: true,
  });

  return (
    <TableContainer>
      <Table>
        <TableHead sx={{ cursor: "pointer" }}>
          {table.getHeaderGroups().map((headerGroup) => (
            <TableRow key={headerGroup.id}>
              {headerGroup.headers.map((header) => (
                <TableCell
                  sx={{ fontWeight: "bold", fontSize: "1.2rem" }}
                  component="th"
                  scope="row"
                  key={header.id}
                  colSpan={header.colSpan}
                >
                  {header.isPlaceholder ? null : (
                    <div
                      {...{
                        onClick: header.column.getToggleSortingHandler(),
                      }}
                    >
                      {flexRender(
                        header.column.columnDef.header,
                        header.getContext()
                      )}
                      {{
                        asc: " ðŸ”¼",
                        desc: " ðŸ”½",
                      }[header.column.getIsSorted() as string] ?? null}
                    </div>
                  )}
                </TableCell>
              ))}
            </TableRow>
          ))}
        </TableHead>
        <TableBody>
          {table.getRowModel().rows.map((row) => (
            <TableRow key={row.id}>
              {row.getVisibleCells().map((cell) => (
                <TableCell component="td" scope="row" key={cell.id}>
                  {flexRender(cell.column.columnDef.cell, cell.getContext())}
                </TableCell>
              ))}
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
};
