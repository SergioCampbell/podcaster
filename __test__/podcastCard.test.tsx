import { PodcastCards } from "@/components/Podcast/PodcastCards";
import { render } from "@testing-library/react";
import { allPodcast } from "../db/allPodcast";

describe("Should render the PodcastCard", () => {
  it("renders the PodcastCard", () => {
    expect(true).toBeTruthy();
  });
  it("Should display the artist name", () => {
    const card = render(
      <PodcastCards allPodcast={allPodcast} loading={false} />
    );
    let tree = card.asFragment();
    expect(tree).toMatchSnapshot();
  });
});
