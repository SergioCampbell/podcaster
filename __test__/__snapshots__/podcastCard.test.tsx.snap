// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the PodcastCard Should display the artist name 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <a
      href="/podcast/1718562878"
      style="color: white; text-decoration: none;"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1vo4off-MuiPaper-root-MuiCard-root"
        data-testid="podcast-artist"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-9pp4f2-MuiAvatar-root"
          >
            <img
              alt="Mortal Sin - NBC News"
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="https://is1-ssl.mzstatic.com/image/thumb/Podcasts126/v4/73/e4/8e/73e48e46-2812-f78f-e571-454e85b59be7/mza_4549916304669920567.jpg/170x170bb.png"
            />
          </div>
          <h5
            aria-label="Mortal Sin - NBC News"
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1q4wj9t-MuiTypography-root"
            data-mui-internal-clone-element="true"
          >
            Mortal Sin - NBC News
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-17eqfq4-MuiTypography-root"
          >
            NBC News
          </p>
        </div>
      </div>
    </a>
  </div>
  <div
    class="MuiBox-root css-0"
  >
    <a
      href="/podcast/1538165808"
      style="color: white; text-decoration: none;"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1vo4off-MuiPaper-root-MuiCard-root"
        data-testid="podcast-artist"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-9pp4f2-MuiAvatar-root"
          >
            <img
              alt="Who shat on the floor at my wedding? - Who shat on the floor at my wedding?"
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="https://is1-ssl.mzstatic.com/image/thumb/Podcasts116/v4/d1/f4/ae/d1f4ae64-32af-d8a5-9531-07bd8bef2a05/mza_3539549108746744251.jpeg/170x170bb.png"
            />
          </div>
          <h5
            aria-label="Who shat on the floor at my wedding? - Who shat on the floor at my wedding?"
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1q4wj9t-MuiTypography-root"
            data-mui-internal-clone-element="true"
          >
            Who shat on the floor at my wedding? - Who shat on the floor at my wedding?
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-17eqfq4-MuiTypography-root"
          >
            Who shat on the floor at my wedding?
          </p>
        </div>
      </div>
    </a>
  </div>
  <div
    class="MuiBox-root css-0"
  >
    <a
      href="/podcast/1719795462"
      style="color: white; text-decoration: none;"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1vo4off-MuiPaper-root-MuiCard-root"
        data-testid="podcast-artist"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-9pp4f2-MuiAvatar-root"
          >
            <img
              alt="On Musk with Walter Isaacson - iHeartPodcasts and Kaleidoscope"
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="https://is1-ssl.mzstatic.com/image/thumb/Podcasts116/v4/e2/30/9c/e2309c86-d54f-1c85-2121-67c7104441df/mza_3219819313662797266.jpg/170x170bb.png"
            />
          </div>
          <h5
            aria-label="On Musk with Walter Isaacson - iHeartPodcasts and Kaleidoscope"
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1q4wj9t-MuiTypography-root"
            data-mui-internal-clone-element="true"
          >
            On Musk with Walter Isaacson - iHeartPodcasts and Kaleidoscope
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-17eqfq4-MuiTypography-root"
          >
            iHeartPodcasts and Kaleidoscope
          </p>
        </div>
      </div>
    </a>
  </div>
  <div
    class="MuiBox-root css-0"
  >
    <a
      href="/podcast/1200361736"
      style="color: white; text-decoration: none;"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1vo4off-MuiPaper-root-MuiCard-root"
        data-testid="podcast-artist"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-9pp4f2-MuiAvatar-root"
          >
            <img
              alt="The Daily - The New York Times"
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="https://is1-ssl.mzstatic.com/image/thumb/Podcasts115/v4/1c/ac/04/1cac0421-4483-ff09-4f80-19710d9feda4/mza_12421371692158516891.jpeg/170x170bb.png"
            />
          </div>
          <h5
            aria-label="The Daily - The New York Times"
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1q4wj9t-MuiTypography-root"
            data-mui-internal-clone-element="true"
          >
            The Daily - The New York Times
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-17eqfq4-MuiTypography-root"
          >
            The New York Times
          </p>
        </div>
      </div>
    </a>
  </div>
  <div
    class="MuiBox-root css-0"
  >
    <a
      href="/podcast/77001367"
      style="color: white; text-decoration: none;"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1vo4off-MuiPaper-root-MuiCard-root"
        data-testid="podcast-artist"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-9pp4f2-MuiAvatar-root"
          >
            <img
              alt="The Ramsey Show - Ramsey Network"
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="https://is1-ssl.mzstatic.com/image/thumb/Podcasts116/v4/2a/db/c7/2adbc7d6-dcde-d8ae-481e-7f4826abf37a/mza_12577097097957624675.jpg/170x170bb.png"
            />
          </div>
          <h5
            aria-label="The Ramsey Show - Ramsey Network"
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1q4wj9t-MuiTypography-root"
            data-mui-internal-clone-element="true"
          >
            The Ramsey Show - Ramsey Network
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-17eqfq4-MuiTypography-root"
          >
            Ramsey Network
          </p>
        </div>
      </div>
    </a>
  </div>
  <div
    class="MuiBox-root css-0"
  >
    <a
      href="/podcast/1322200189"
      style="color: white; text-decoration: none;"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1vo4off-MuiPaper-root-MuiCard-root"
        data-testid="podcast-artist"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-9pp4f2-MuiAvatar-root"
          >
            <img
              alt="Crime Junkie - audiochuck"
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="https://is1-ssl.mzstatic.com/image/thumb/Podcasts126/v4/8c/35/04/8c350430-2fbf-98d0-0a25-00b76550ffeb/mza_13445204151221888086.jpg/170x170bb.png"
            />
          </div>
          <h5
            aria-label="Crime Junkie - audiochuck"
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1q4wj9t-MuiTypography-root"
            data-mui-internal-clone-element="true"
          >
            Crime Junkie - audiochuck
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-17eqfq4-MuiTypography-root"
          >
            audiochuck
          </p>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
